Content.Server/Clothing/NightVisionSystem.cs

```diff
@@ -0,0 +1,36 @@
+using Content.Shared.Clothing;
+using Content.Shared.Alert;
+using Content.Shared.Inventory.Events;
+
+namespace Content.Server.Clothing;
+
+/// <summary>
+/// Made by BL02DL from _LostParadise
+/// </summary>
+
+public sealed class NightVisionSystem : SharedNightVisionSystem
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<NightVisionComponent, GotEquippedEvent>(OnGotEquipped);
+        SubscribeLocalEvent<NightVisionComponent, GotUnequippedEvent>(OnGotUnequipped);
+    }
+
+    private void OnGotUnequipped(EntityUid uid, NightVisionComponent component, GotUnequippedEvent args)
+    {
+        if (args.Slot == "eyes")
+        {
+            UpdateNightVisionEffects(args.Equipee, uid, false, component);
+        }
```

Comment:
```suggestion
        if (args.Slot == component.Slot)
            UpdateNightVisionEffects(args.Equipee, uid, false, component);
```
You also need to make a [DataField] public string Slot = "eyes"; entry in the NightVisionComponent for this to work.